apps/web/src/components/chat/assistent-chat-message.tsx

```diff
@@ -0,0 +1,375 @@
+                  messageId={message.id}
+                  index={index}
+                />
+              );
+            } else if (part.type === "text") {
+              return (
+                <MemoizedMarkdown
+                  key={index}
+                  part={part}
+                  messageId={message.id}
+                />
+              );
+            } else if (
+              part.type === "file" &&
+              "mediaType" in part &&
+              part.mediaType?.startsWith("image/")
+            ) {
+              // Handle image files
+              return <ImagePart part={part} key={index} />;
+            } else if (part.type === "step-start") {
+              // Step start parts are typically not rendered visually
+              return null;
+            } else if (isToolPart(part)) {
+              // Handle individual tool parts
+              return <ToolMessage key={index} part={part} />;
+            }
+            return null;
+          })}
+
+          {attachments && attachments.length > 0 && (
+            <div className="mt-2 flex flex-wrap gap-2">
+              {attachments.map(
+                (attachment: MessageAttachment, index: number) => (
+                  <AttachmentCard
+                    key={`${message.id}-${index}`}
+                    attachment={attachment}
+                  />
+                ),
+              )}
+            </div>
+          )}
+
+          {hasTextContent && (
+            <div className="mt-2 flex w-full items-center justify-end gap-2 text-xs text-muted-foreground opacity-0 pointer-events-none transition-all duration-200 group-hover:opacity-100 group-hover:pointer-events-auto">
+              <div>
+                <Button
+                  onClick={handleCopyMessage}
+                  variant="ghost"
+                  size="xs"
+                  className="text-muted-foreground hover:text-foreground p-0 h-auto"
+                >
+                  <Icon name="content_copy" className="text-sm" />
+                </Button>
+              </div>
+            </div>
+          )}
+        </div>
+      ) : showDots ? (
+        <Dots />
+      ) : null}
+    </div>
+  );
+}
+
+function ImagePart({ part }: { part: FileUIPart }) {
+  return (
+    <a
+      href={part.url}
+      target="_blank"
+      rel="noopener noreferrer"
+      className="relative group flex items-center gap-2 rounded-xl"
+    >
+      <div className="relative">
+        <img
+          src={part.url}
+          alt={part.filename || "Uploaded image"}
+          className="rounded-lg max-h-[300px] object-cover"
+        />
+      </div>
+    </a>
+  );
+}
+
+function AttachmentCard({ attachment }: { attachment: MessageAttachment }) {
+  const contentType =
+    attachment.contentType ||
+    (attachment.url.startsWith("data:")
+      ? attachment.url.split(":")[1]?.split(";")[0]
+      : undefined) ||
+    "application/octet-stream";
+  const isImage = contentType.startsWith("image/");
```

Comment:
Do we really need all of this logic?

I'd expect for us to already have that somewhere